components/tools/OmeroJava/test/integration/ThumbnailStoreTest.java

```diff
@@ -229,4 +230,73 @@ public void testGetThumbnailAfterReset() throws Exception {
         }
         svc.close();
     }
+
+    /**
+     * Test that thumbnails can be retrieved from multiple groups at once.
+     * @throws Throwable unexpected
+     */
+    @Test
+    public void testGetThumbnailsMultipleGroups() throws Throwable {
+        final byte[] thumbnail;
+        final long pixelsIdα, pixelsIdβ;
+        ThumbnailStorePrx svc = null;
+
+        /* create a fake image file */
+        final File file = File.createTempFile(getClass().getSimpleName(), ".fake");
+        file.deleteOnExit();
+
+        try {
+            /* import the image as one user in one group and get its thumbnail */
+            pixelsIdα = importFile(importer, file, "fake", false).get(0).getId().getValue();
+            svc = factory.createThumbnailStore();
+            svc.setPixelsId(pixelsIdα);
+            thumbnail = svc.getThumbnailByLongestSide(null);
+        } finally {
+            if (svc != null) {
+                {
+                    svc.close();
+                    svc = null;
+                }
+            }
+        }
+
+        setUpNewUserWithImporter();
+
+        try {
+            /* import the image as another user in another group and get its thumbnail */
+            pixelsIdβ = importFile(importer, file, "fake", false).get(0).getId().getValue();
+            svc = factory.createThumbnailStore();
+            svc.setPixelsId(pixelsIdβ);
+            Assert.assertEquals(svc.getThumbnailByLongestSide(null), thumbnail);
+        } finally {
+            if (svc != null) {
+                {
+                    svc.close();
+                    svc = null;
+                }
+            }
+        }
+
+        final Map<Long, byte[]> thumbnails;
+
+        try {
+            /* use all-groups context to fetch both thumbnails at once */
+            final List<Long> pixelsIdsαβ = ImmutableList.of(pixelsIdα, pixelsIdβ);
+            final Map<String, String> allGroupsContext = ImmutableMap.of("omero.group", "-1");
+            svc = factory.createThumbnailStore();
+            thumbnails = svc.getThumbnailByLongestSideSet(null, pixelsIdsαβ, allGroupsContext);
```

Comment:
`OmeroInterceptor.neededRight` does look like it, thank you. Maybe I was looking in just `BasicACLVoter` before. Interesting to see in that function that a `RenderingDef` can be linked to a `RenderingDef`.